### PR TITLE
 finished QuestionBox-ui

### PR DESCRIPTION
**Pull request by Tante-Didi**
still i need to remove the black border color that i dont know how to stop it

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "editor.formatOnSave": true
```
Comment (pini85) [inline .vscode/settings.json:2]: please dont commit vs code settings

File: src/components/ui/QuestionBox/QuestionBox.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React, { useState } from "react";
```
Comment (pini85) [inline src/components/ui/QuestionBox/QuestionBox.jsx:1]: Our feature was only to create a Question button component.
Our component here is doing too much. It is also handling the logic and also displaying the button. 
We should also decouple our components and it should do only one thing, and do one thing really well. 
When our component does multiple things, thats when the potential bugs can occur. 
We should create a component that accepts via props a color, and also that component should accept a callback function when you click on it. Later on we will do the logic.
This way we have a truly reusable component that we can put anywhere in our application.

File: src/components/ui/QuestionBox/QuestionBox.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  const questionData = dummyData[0].questions[0];
+
+  const handleAnswerClick = (index) => {
+    if (isAnswered) return; // If already answered, do nothing
+
+    const updatedColors = [...answerColors];
+
+    if (index === questionData.correctAnswer) {
+      updatedColors[index] = "var(--color-for-correct-answer)";
```
Comment (pini85) [inline src/components/ui/QuestionBox/QuestionBox.jsx]: we should rename our css variable colors to be reusable. Green, blue, red etc. For example the red and green in your assignment is the same red and green for the toast message

File: src/components/ui/QuestionBox/QuestionBox.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      updatedColors[index] = "var(--color-for-wrong-answer)";
+      updatedColors[questionData.correctAnswer] = "var(--color-for-correct-answer)";
+    }
+
+    setAnswerColors(updatedColors);
+    setIsAnswered(true);
+  };
+
+  return (
+    <StyledQuestionBox>
```
Comment (pini85) [inline src/components/ui/QuestionBox/QuestionBox.jsx]: One other convention which I didnt share with you is that for styled components it is easier to read and understand that they are styled components if they have a convention prefix as S
you do it as follows:
import * as S from "./foo.styles.js
<S.QuestionBox>
Please see the Button.jsx for reference

File: src/components/ui/QuestionBox/QuestionBox.style.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  height: 800px;
+
+  justify-content: center;
+  font-family: var(--font-rubik-regular);
+  font-weight: 500;
+  font-size: 12px;
+  line-height: 24px;
+  border-radius: 8px;
+  border-color: transparent;
+  color: var(--font-color--Answer--QuestionBox);
```
Comment (pini85) [inline src/components/ui/QuestionBox/QuestionBox.style.js]: this variable name is confusing

File: src/pages/QuestionPage.jsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import QuestionBox from "../components/ui/QuestionBox";
+
+const QuestionPage = () => {
+  return (
+    <div>
```
Comment (pini85) [inline src/pages/QuestionPage.jsx]: semantically it should be main not div. It will be the main element of the page (once we do the page)

**Review by pini85 — CHANGES_REQUESTED** [review]

File: src/components/ui/QuestionBox/QuestionBox.jsx (hunk 3 above)
Comment (Tante-Didi) [reply src/components/ui/QuestionBox/QuestionBox.jsx]: ok thx,
I change it :
image

File: .vscode/settings.json (hunk 1 above)
Comment (Tante-Didi) [reply .vscode/settings.json:2]: I am not familair with this change. however, next time before I will commit changes, I will do git status

File: src/components/ui/QuestionBox/QuestionBox.style.js (hunk 5 above)
Comment (Tante-Didi) [reply src/components/ui/QuestionBox/QuestionBox.style.js]: no problem, was changed as you suggested, just the color name :)

image

File: src/pages/QuestionPage.jsx (hunk 6 above)
Comment (Tante-Didi) [reply src/pages/QuestionPage.jsx]: THX ,  
image

File: src/components/ui/QuestionBox/QuestionBox.jsx (hunk 4 above)
Comment (Tante-Didi) [reply src/components/ui/QuestionBox/QuestionBox.jsx]: do u mean ,like that  
in the QuestionBox.style .js :
image


image

QuestionBox.jsx :
image

**Pull request merged**